streaming/src/main/scala/org/apache/spark/streaming/receiver/Receiver.scala

```diff
@@ -107,9 +107,6 @@ abstract class Receiver[T](val storageLevel: StorageLevel) extends Serializable
    */
   def onStop()
 
-  /** Override this to specify a preferred location (hostname). */
-  def preferredLocation : Option[String] = None
```

Comment:
I have tried, but didn't find a good way to pass the argument through `ReceiverInputDStream` to this by adding some methods.
IMO, `ReceiverInputDStream` is also an interface for user, user can overide preferredLocation (`ReceiverInputDStream`) in their own implementation.